src/crafting.cpp

```diff
@@ -859,6 +860,24 @@ comp_selection<item_comp> player::select_item_component( const std::vector<item_
     return selected;
 }
 
+// Prompts player to empty all newly-unsealed containers in inventory
+// Called after something that might have opened containers (making them buckets) but not emptied them
+void empty_buckets( player &p )
+{
+    // First grab (remove) all items that are non-empty buckets and not wielded
+    auto buckets = p.remove_items_with( [&p]( const item &it ) {
+        return it.is_bucket_nonempty() && &it != &p.weapon;
+    }, INT_MAX );
+    for( auto &it : buckets ) {
+        for( const item &in : it.contents ) {
+            drop_or_handle( in, p );
+        }
+
+        it.contents.empty();
```

Comment:
Good catch - it was obviously supposed to be `clear`.

No, it will not prefer the open one. Fixing that would require either doing a two-pass consumption (ugly) or overhauling the crafting selection to support priorities.